modin/core/io/text/excel_dispatcher.py

```diff
@@ -153,7 +153,7 @@ def _read(cls, io, **kwargs):
             if index_col is not None:
                 column_names = column_names.drop(column_names[index_col])
 
-            if not all(column_names):
+            if not all(column_names) or kwargs.get("usecols"):
```

Comment:
just interesting, why would we ever want to compute columns ourselves instead of using pandas to read proper metadata?

I'm asking because if there's a solid reason for not using pandas here then I believe we should manually process the `usecols` parameter. It shouldn't be that hard to get a subset of `column_names` ourselves using data from `usecols` param.

Reply:
@dchigarev In any case, without this change, the code does not work correctly. If necessary, it will be possible in the future to try to handle this parameter ourselves in order to get a performance boost.